### ISSUE TITLE
PL/SQL parse error when use BULK COLLECT INTO

### ISSUE THREAD

**Issue by raidzin** [sqlfluff/sqlfluff#7073]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Get `Found unparsable section: 'COLLECT'`  when using [Oracle Nested Tables](https://docs.oracle.com/en/database/oracle/oracle-database/23/lnpls/nested-tables.html)

### Expected Behaviour

No prasing error

### Observed Behaviour

sqlfluff output for `example.sql`:

```
== [example.sql] FAIL
L:  20 | P:   5 | LT09 | Select targets should be on a new line unless there is
(servicedesk) PS C:\Users\goncharukak\PycharmProjects\ServiceDeskNews> sqlfluff lint ttt.sql
L:  23 | P:   1 | LT02 | Expected indent of 12 spaces. [layout.indent]
                       | [aliasing.column]
L:  23 | P:   5 | CP02 | Unquoted identifiers must be consistently lower case.
                       | [capitalisation.identifiers]
L:  23 | P:   5 | RF04 | Keywords should not be used as identifiers.
                       | [references.keywords]
L:  23 | P:   9 | LT02 | Expected line break and indent of 4 spaces before
                       | 'COLLECT'. [layout.indent]
L:  23 | P:  10 |  PRS | Line 23, Position 10: Found unparsable section:
                       | 'COLLECT'
WARNING: Parsing errors found and dialect is set to 'oracle'. Have you configured your dialect correctly?
```

### How to reproduce

`example.sql`

```sql
CREATE OR REPLACE PACKAGE example_package AS

TYPE foo_bar IS RECORD (foo varchar2(100), bar varchar2(100));
TYPE foo_bars IS TABLE OF foo_bar;

FUNCTION get_foo_bars(
    foo_bar_id number
) RETURN foo_bars;

END example_package;

CREATE OR REPLACE PACKAGE BODY example_package AS

FUNCTION get_foo_bars(
    foo_bar_id number
) RETURN foo_bars
IS
selected_foo_bars foo_bars;
BEGIN
    SELECT
        foobar_table.foo,
        foobar_table.bar
    BULK COLLECT INTO selected_foo_bars
    FROM foobar_table
    WHERE foobar_table.id = foo_bar_id; -- noqa: RF03

    RETURN selected_foo_bars;
END;

END example_package;

```

### Dialect

oracle

### Version

sqlfluff, version 3.4.2
python 3.12

### Configuration

[tool.sqlfluff.core]
dialect = "oracle"

[tool.sqlfluff.rules.capitalisation.keywords]
capitalisation_policy = "upper"

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
